### DISCUSSION TITLE
Issue with transcribing sh as zh in Czech and transcribing z as z in general

### DISCUSSION THREAD

**Discussion by michal-bida** [espeak-ng/espeak-ng#1951] [Q&A] [answered]
Hello. 

There seems to be an issue in Czech phonemization as, e.g. word rozhlas is phonemized incorrectly. The rules are matched correctly but the output instead of R'ozhlas shows R'oshlas...
```
❯ espeak -v cs -X "rozhlas"
Translate 'rozhlas'
  1	r        [R]

  1	o        [o]

  1	z        [z]

  1	h        [h]

  1	l        [l]

  1	a        [a]

  1	s        [s]

 R'oshlas
```

Where did this z to s come from?

Even when I defined a new group in Czech grammar called sh as these two characters have special way of pronunciation. 
In this group, sh is by default transcribed to sx (that is correct - that is the more common Czech pronunciation). 
But there are exceptions to this rules I have listed. Though even while being matched correctly, they are not transcribed as zh but sh. In Czech sh is never pronounced as sh - it is either pronounced as sx or zh. 

Example:

```
.group sh
        sh         sx  //make default the more common pronunciation
        sh (ora    zh
        sh (ůry    zh
        sh (luk    zh  //shluknout, shluk
        sh (lad    zh  //shladit
        sh (oblo   zh  //shoblovat
        sh (oufo   zh  //shoufovat
    di) sh (armon  zh  //disharmonie

```

```
❯ ESPEAK_DATA_PATH=`pwd` LD_LIBRARY_PATH=src:${LD_LIBRARY_PATH} src/espeak-ng -v cs -X "shora"
Translate 'shora'
 81     sh (ora  [zh]
 36     sh       [sx]
  1     s        [s]

  1     o        [o]

  1     r        [R]

  1     a        [a]

sh'oRa
```

Thanks for any help.

**Comment by jaacoppi** [comment, marked as answer]
Root cause might be in the phoneme substitution. See phsource/phonemes and find which tables are in use for Czech.

The phoneme substitutions listed there will happen after the transcribing part, so unfortunately espeak-ng -X won't show them.

**Reply by michal-bida** [reply]
@jaacoppi Thanks a lot for help!

I found the following transcriptions in the phonemes file that seems relevant:

```
phoneme z
phoneme Z // plea**s**ure
phoneme z.
phoneme z;
phoneme Z;
```

In detail roughly this is happening:
```
phoneme z.
  vcd pla frc sib
  ipa ʐ
  voicingswitch s.
  lengthmod 6
  Vowelin  f1=0  f2=1800 -100 300  f3=-300 80
  Vowelout f1=2  f2=1800 -100 300  f3=-300 80 brk

  IF KlattSynth THEN
    IF nextPh(isPause2) THEN
      FMT(klatt/zh_) addWav(vocw/zh_rfx)
    ENDIF
    FMT(klatt/zh) addWav(vocw/zh_rfx)
  ENDIF

  IF nextPh(isPause2) THEN
    FMT(voc/zh_) addWav(vocw/zh_rfx)
  ENDIF
  FMT(voc/zh) addWav(vocw/zh_rfx)
endphoneme
```

I guess `voicingswitch s` might be the reason of z -> s transcription?

 When I do this change:
 ```
 phoneme z
  vcd alv frc sib
  voicingswitch z
  ```
Then the czech phonemization works correctly. E.g.:
`ESPEAK_DATA_PATH=`pwd` LD_LIBRARY_PATH=src:${LD_LIBRARY_PATH} src/espeak-ng -v cs -X "shluk"` -> zhl'uk`

My only concern is that:
- I have no idea what I am doing - how will the other languages be affected? Is this even correct concerning the code seems to affect speech synthesis? 
- Is this something that could be merged? Probably not? Probably it needs to be added into ph_czech file as a overriding rule right?

One other let's say more general question is - in general, is this behavior (s->z universally) a bug or feature of espeak? I understand that it works as intended by default for English and if some other languages require other handling they should override this behavior... Right?
